python/ray/tests/test_runtime_env.py

```diff
@@ -441,7 +441,7 @@ def test_two_node_uri(two_node_cluster, working_dir, client_mode):
         pkg_uri = runtime_env.Protocol.PIN_GCS.value + "://" + pkg_name
         runtime_env.create_project_package(working_dir, [], [], tmp_file.name)
         runtime_env.push_package(pkg_uri, tmp_file.name)
-        runtime_env = f"""{{ "working_dir_uri": "{pkg_uri}" }}"""
+        runtime_env = f"""{{ "uris": ["{pkg_uri}"] }}"""
```

Comment:
I think we should actually remove the separation of "working_dir" and "working_dir_uri". We can just have "working_dir", and the value can either be a string path, single URI, or JSON dict that describes how to reconstruct the working dir (e.g., {"type": "base_plus_delta", "base": "s3://file.zip", "delta": "s3://delta.zip"}). This would allow us to support multiple types of encoding.

Reply:
Ah, it's a good idea. Probably we should mark working_dir as :: `file://` so that we know it's a local dir.

As for the path, we can introduce some schema, like "file://dir@add", "s3://key@base", "s3://key@add", "s3://key@del", "HTTP://key@add"

If it's "file://dir" it'll rewrite to other forms for uploading and downloading later.

In this way, we actually can store the training data(100GB?) in the remote node completely instead of having that in the local node.

Reply:
I think it's a bit weird to try to encode too much into the URIs, rather we should just put a json dict with type field. Then, different plugins can interpret the contents of the dict in plugin specific ways, which would be more idiomatic python.

Reply:
That's right. This is actually an internal interface. It is like this: everything will be compiled into this Uris list and will be shipped to all workers. Maybe I should put it and _uris instead.

For things like working dir or other kind of thing, I'll view it one layer higher than Uris. Basically, we'd like to have an unified way to manage these resources in the low level (gcs, Raylet) and adding new types of resources doesn't need to update lower level code.